tests/tracer/test_gitmetadata.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+"""
+tests for git metadata embedding and processing.
+"""
+import glob
+import os
+import subprocess
+
+import pytest
+
+import ddtrace
+from tests.subprocesstest import run_in_subprocess
+from tests.utils import DummyWriter
+from tests.utils import TracerTestCase
+
+
+@pytest.fixture
+def preapare_test_env(example):
+    subprocess.check_output("python setup.py bdist_wheel", shell=True)
+    pkgfile = glob.glob(os.path.join(example, "dist", "*.whl"))[0]
+
+    os.environ["SHA_VALUE"] = subprocess.check_output("git rev-parse HEAD", shell=True).decode("utf-8").strip()
+
+    envdir = os.path.join(example, "run_env_dir")
+    cwd = os.getcwd()
+    python_path = os.getenv("PYTHONPATH", None)
+    try:
+        subprocess.check_output("pip install --target=" + envdir + " " + pkgfile, shell=True)
+        os.chdir(envdir)
+        os.environ["PYTHONPATH"] = os.getenv("PYTHONPATH", "") + os.pathsep + envdir
+        yield
+    finally:
+        os.chdir(cwd)
+        if python_path is not None:
+            os.environ["PYTHONPATH"] = python_path
+
+
+@pytest.mark.usefixtures("preapare_test_env")
+class GitMetadataTestCase(TracerTestCase):
```

Comment:
do we need any failure case test cases?

e.g. `DD_GIT_COMMIT_SHA` set but not `DD_GIT_REPOSITORY_URL`, or malformed values in the env vars?

also, what about an example that only uses `DD_TAGS`?

Reply:
What about from a package that doesn't have `Project-URL` set?

Reply:
> e.g. DD_GIT_COMMIT_SHA set but not DD_GIT_REPOSITORY_URL, or malformed values in the env vars?

We do not do these checks in any of the tracers, the values go as they are and are checked by the backend if necessary.

Reply:
> What about from a package that doesn't have Project-URL set?

Test added